evaluation/EDA/run_infer.py

```diff
@@ -205,13 +205,15 @@ def process_instance(
         specified_llm_config = get_llm_config_arg(args.llm_config)
         if specified_llm_config:
             config.llm = specified_llm_config
+    config.max_iterations = args.max_iterations
```

Comment:
@neubig I think the main reason/place for involving global `config` is actually to pass the correct `max_iterations` value so that the `controller` will stop correctly. I think the eval is currently broken if this is not set - it will follow the `max_iterations` set in `config.toml`, rather than the one specified in the arguments.

Let me see if there's other easier way to get this fixed here

Reply:
I think it must be because we're now not running `main`, but `run_agent_controller`, which assumes the args have already been parsed and set. I see that was the solution you found, too. And I wonder if there are others... When I get to my computer I should test eval dir.

Reply:
Yeah... we need some serious refractor on the config thing to clear things up